debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/oracle/generated/PlSqlParser.g4

```diff
@@ -2157,7 +2157,239 @@ access_driver_type
 
 external_table_data_props
     : DEFAULT DIRECTORY quoted_string
-    | LOCATION
+    | ACCESS PARAMETERS '(' et_oracle_loader | et_oracle_datapump | et_oracle_hdfs_hive | USING CLOB subquery ')'
+    | LOCATION '(' et_directory_spec? et_location_specifier (',' et_directory_spec? et_location_specifier )* ')'
+    ;
+
+et_oracle_loader
+    : comments_oracle_loader? record_format_info? field_definitions? column_transforms?
+    ;
+
+et_oracle_datapump
+    : ENCRYPTION (ENABLED | DISABLED)
+      | ( NOLOGFILE | LOGFILE et_directory_spec? et_file_spec )
+      | COMPRESSION ( ENABLED (BASIC |LOW | MEDIUM | HIGH)? | DISABLED)?
+      | HADOOP_TRAILERS (ENABLED | DISABLED) VERSION (COMPATIBLE | LATEST | quoted_string )
+      | comments_oracle_datapump
+    ;
+
+et_oracle_hdfs_hive
+    : COM_ORACLE_HDFS_HIVE_BIGDATA_COLMAP ( '=' | ':' ) (
+        colmap_entry | ('[' colmap_entry ( ',' colmap_entry)* ']'))
+    | COM_ORACLE_HDFS_HIVE_BIGDATA_ERROROPT ( '=' | ':' ) (
+        error_element | ('[' error_element ( ',' error_element)* ']' ))
+    | COM_ORACLE_HDFS_HIVE_BIGDATA_FIELDS ( '=' | ':' ) (
+        field_spec et_data_type
+        ( COMMENT col_comment )? (',' COMMENT col_comment )*? )
+    | COM_ORACLE_HDFS_HIVE_BIGDATA_FILEFORMAT ( '=' | ':' ) (
+        SEQUENCEFILE
+        | TEXTFILE
+        | RCFILE
+        | ORC
+        | PARQUET
+        | INPUTFORMAT quoted_string OUTPUTFORMAT quoted_string )
+    | COM_ORACLE_HDFS_HIVE_BIGDATA_EXEC ( '=' | ':' ) (
+        et_directory_spec? et_file_spec )
+    | COM_ORACLE_HDFS_HIVE_BIGDATA_QC ( '=' | ':' ) (
+        et_directory_spec? et_file_spec )
+    | COM_ORACLE_HDFS_HIVE_BIGDATA_OVERFLOW ( '=' | ':' ) (
+        overflow_element | ('[' overflow_element ( ',' overflow_element)* ']'))
+    | COM_ORACLE_HDFS_HIVE_BIGDATA_ROWFORMAT ( '=' | ':' ) (
+        DELIMITED?
+        ( FIELDS TERMINATED BY CHARACTER ( ESCAPED BY CHARACTER )
+        | COLLECTION ITEMS TERMINATED BY CHARACTER
+        | MAP KEYS TERMINATED BY CHARACTER
+        | LINES TERMINATED BY CHARACTER
+        | NULL_ DEFINED AS CHARACTER
+        )
+        | SERDE quoted_string ( WITH SERDEPROPERTIES ( quoted_string '=' quoted_string ( ',' quoted_string '=' quoted_string)*))? )
+    | COM_ORACLE_HDFS_HIVE_BIGDATA_TABLENAME ( '=' | ':' ) (
+        ( DEFAULT | quoted_string ) '.' quoted_string )
+    ;
+
+colmap_entry
+    : LEFT_CURLY_PAREN ET_COL ':' quoted_string ',' ET_FIELD ':' quoted_string RIGHT_CURLY_PAREN
```

Comment:
I noticed here that the `':' quoted_string` gets picked by the Lexer as `BINDVAR`, which makes sense.  @jpechane or @gunnarmorling do either of you have a suggestion on how to best handle this use case?  We have a number of places in these rules where this syntax is used but gets recognized as a bind variable lexer token.

Reply:
I'm not quite sure I follow. Can you provide a complete statement which gets parsed into a wrong AST, and describe the error (wrong rule application)?

That said, does it matter from our perspective? Keep in mind we don't care about 95% of the grammar, we only are interested in a small subset of information, like column types. All the handling of partition clauses and what not is just a chore as a grammar must be complete by definition. So I'd say while surely not attractive from a general correctness perspective, I personally wouldn't mind if there are some wrong parsing results in AST parts which we don't care about. At least I wouldn't spend too much time on it.

Reply:
So using Anisha's `colmap_entry` as an example, lets take a look at [this SQL](https://github.com/debezium/debezium/blob/78bca37b0537e7b0d40bf1d73fec469e240ae059/debezium-ddl-parser/src/test/resources/oracle/examples/ddl_create.sql) Line 147 in the example files.  We have defined a set of access parameters.  What I find is that when we parse:
```
com.oracle.bigdata.tableName: {"col":"ITEM_CNT", "field":"order_line_item_count"}
```
We don't recognize `:"ITEM_CNT"` or `:"order_line_item_count"` as a token stream of `:` and `"<string value>"` but instead the Lexer recongizes as `BINDVAR` because there is a Lexer rule that says if the SQL is a `:` followed by a string then it is to be treated like a JDBC bind variable declaration of `:abc`. 

Now when I did this review, it didn't occur to me, but looking back, I'm curious if we should adjust `BINDVAR` and remove the `':' DELIMITED_ID` rule entirely.  I don't think it makes sense that a `BINDVAR` is recognized as `:"<identifier>"` but only if its `:<identifier>` (no quotes).

That might fix the `colmap_entry` issue if we did that, but that's not the only place this happens.  Another example is also Line 80 in the same SQL example file where we have field declarations with start/length tuples:
```
     (FIELDS RTRIM
            (first_name (1:15) CHAR(15),
             last_name (*:+20),
             year_of_birth (36:39),
             phone (40:52),
             area_code (*-12: +3),
             exchange (*+1: +3),
             extension (*+1: +4)))
```
Here we recognize `:15` for `first_name` as a `BINDVAR` rather than `:` followed by an `UNSIGNED_INTEGER`.  This is again because the Lexer rule takes precedence over any rules in the parser if I understand ANTLR correctly.

So on the parser side, this means that in all these situations where we have used `:` followed by `UNSIGNED_INTEGER` we need to use `BINDVAR` so that it matches; however, that seems dubious to me from a grammar perspective.

I get we don't really care about these aspects; however, this makes it a bit tough to stay aligned with the upstream grammar and contribute to it if we begin to have such heavy deviations I think too.

Maybe for now we just define the rules to accept `BINDVAR` so that the rules pass and the SQL is parsed and we then add a follow-up to see if we could revisit this in a more appropriate way.  You would never apply a bind variable in these use cases where we see this happen so I wonder if we need to consider moving `BINDVAR` into the parser stage and avoid it being emitted as a lexicon in the lexical analysis.